services/galley/src/Galley/API/Clients.hs

```diff
@@ -111,12 +112,20 @@ rmClientH ::
   UserId ::: ClientId ->
   Sem r Response
 rmClientH (usr ::: cid) = do
-  lusr <- qualifyLocal usr
-  let nRange1000 = toRange (Proxy @1000) :: Range 1 1000 Int32
-  firstConvIds <- Query.conversationIdsPageFrom lusr (GetPaginatedConversationIds Nothing nRange1000)
-  goConvs nRange1000 firstConvIds lusr
-
-  E.deleteClient usr cid
+  clients <- E.getClients [usr]
+  if (cid `elem` clientIds usr clients)
+    then do
+      lusr <- qualifyLocal usr
+      let nRange1000 = toRange (Proxy @1000) :: Range 1 1000 Int32
+      firstConvIds <- Query.conversationIdsPageFrom lusr (GetPaginatedConversationIds Nothing nRange1000)
+      goConvs nRange1000 firstConvIds lusr
+      E.deleteClient usr cid
+    else
+      P.debug
+        ( field "user" (idToText usr)
+            . field "client" (clientToText cid)
+            . msg (val "rmClientH: client already gone")
```

Comment:
While at it, why not rename the function too? We have the "H" suffix (I guess "H" for handler) in the old `wai` routes days. We have been dropping them in the migration to Servant.

```suggestion
            . msg (val "rmClient: client already gone")
```

Reply:
I would generally agree, but cosmetics should be done on `develop` only. I try to keep the PRs added to two branches to the minimal change with maximal code overlap.